internal/ddt/node.go

```diff
@@ -346,6 +346,28 @@ func (n *Node) Version() string {
 func (n *Node) Asset(name string) (*NodeAsset, error) {
 	path := filepath.Join(n.Path, name)
 
+	files, err := ioutil.ReadDir(n.Path)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// read files in current directory
+	for _, f := range files {
+		if fileShouldBeExcluded(f) {
+			continue
+		}
+
+		// if the filename with stripped ordernumber matches the given input return the first match
+		if removeOrderNumber(f.Name()) ==  name {
+			return NewNodeAsset(
+				filepath.Join(n.Path, f.Name()),
+				filepath.Join(n.URL(), f.Name()),
+				n.metaDB,
+			), nil
+		}
+	}
+
```

Comment:
Instead of extracting common functionality from `Assets()`, you can instead just use that function directly. Yes, this adds another loop, but the performance hit is minimal and I favor that over code clarity.

i.e.:

```
for _, a := range n.Assets() {
   // do comparison with both sides normalized by removeOrderNumber
}
```